cpp/include/raft/neighbors/sample_filter.cuh

```diff
@@ -19,10 +19,10 @@
 #include <cstddef>
 #include <cstdint>
 
-namespace raft::neighbors::ivf_pq::detail {
+namespace raft::neighbors::filtering {
 
 /* A filter that filters nothing. This is the default behavior. */
-struct NoneSampleFilter {
+struct NoneIvfSampleFilter {
```

Comment:
Since this is a breaking change anyway, could you please change the naming to the snake case?
In fact, we have a [clang-tidy config](https://github.com/rapidsai/raft/blob/42c9c180f2dcdb1437c45bfb0c753db49ec0a732/cpp/.clang-tidy#L89-L90) for this purpose, but it's not enforced in CI because there's still a lot of violations in the codebase.

Reply:
done